README.md

```diff
@@ -2,243 +2,146 @@
-...
-```
+## Usage
+SR-IOV CNI networks are commonly configured using Multus and SR-IOV Device Plugin using Network Attachment Definitions. More information about configuring Kubernetes networks using this pattern can be found in the [Multus configuration reference document.](https://intel.github.io/multus-cni/doc/configuration.html)
 
-To change the number of VFs reset the number to 0 then set the needed number
+A Network Attachment Definition for SR-IOV CNI takes the form:
 
 ```
-echo 0 > /sys/class/net/enp2s0f0/device/sriov_numvfs
-echo 8 > /sys/class/net/enp2s0f0/device/sriov_numvfs
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-net1
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/intel_sriov_netdevice
+spec:
+  config: '{
+  "type": "sriov",
+  "cniVersion": "0.3.1",
+  "name": "sriov-network",
+  "ipam": {
+    "type": "host-local",
+    "subnet": "10.56.217.0/24",
+    "routes": [{
+      "dst": "0.0.0.0/0"
+    }],
+    "gateway": "10.56.217.1"
+  }
+}'
 ```
 
-Note: In case spoofchk is enabled for VF, a valid administrative MAC needs to be specified.
+The `.spec.config` field contains the configuration information used by the SR-IOV CNI.
 
-## Configuration reference
-### Main parameters
-* `name` (string, required): the name of the network
-* `type` (string, required): "sriov"
-* `deviceID` (string, required): A valid pci address of an SRIOV NIC's VF. e.g. "0000:03:02.3"
-* `vlan` (int, optional): VLAN ID to assign for the VF. Value must be in the range 0-4094 (0 for disabled, 1-4094 for valid VLAN IDs).
-* `vlanQoS` (int, optional): VLAN QoS to assign for the VF. Value must be in the range 0-7. This option requires `vlan` field to be set to a non-zero value. Otherwise, the error will be returned.
-* `mac` (string, optional): MAC address to assign for the VF
-* `ipam` (dictionary, optional): IPAM configuration to be used for this network.
-* `spoofchk` (string, optional): turn packet spoof checking on or off for the VF
-* `trust` (string, optional): turn trust setting on or off for the VF
-* `link_state` (string, optional): enforce link state for the VF. Allowed values: auto, enable, disable. Note that driver support may differ for this feature. For example, `i40e` is known to work but `igb` doesn't.
-* `min_tx_rate` (int, optional): change the allowed minimum transmit bandwidth, in Mbps, for the VF. Setting this to 0 disables rate limiting. The min_tx_rate value should be <= max_tx_rate. Support of this feature depends on NICs and drivers.
+### Basic configuration parameters 
 
-* `max_tx_rate` (int, optional): change the allowed maximum transmit bandwidth, in Mbps, for the VF. 
-Setting this to 0 disables rate limiting. 
+The following parameters are generally outside the scope of SR-IOV CNI configuration, though (with the exception of ipam) they need to be included in the config.
```

Comment:
This `outside the scope of SR-IOV CNI` is a little confusing to me, is this to say these options are generic CNI parameters that are not specific to SR-IOV CNI?

Reply:
Yeah - I'll change this to reflect that more clearly